app/Main.hs

```diff
@@ -133,11 +133,11 @@ eventHandler = \case
         :: MonadIO m
         => Text
         -> Next (VgrepT AppState m Redraw)
-    handleFeedResult line = Continue $ do
-        expandedLine <- expandLineForDisplay line
-        case parseLine expandedLine of
-            Just l  -> zoom results (feedResult l)
-            Nothing -> pure Unchanged
+    handleFeedResult line = Continue $ case parseLine line of
+        Just l  -> do
+            l' <- traverseOf (lineReference . lineText) expandFormattedLine l
+            zoom results (feedResult l')
```

Comment:
Apart from the fact that `traverseOf` is kind of ill-named, I think it is pretty clear what this does. `traverseOf` is the monadic variant of `over` that takes a monadic action and applies it to all traversed elements, just like `over` does with a pure function.

And, interestingly enough, `traverseOf = id`, so we could even omit the `traverseOf` here.